crates/red_knot_python_semantic/src/types.rs

```diff
@@ -5904,6 +5920,15 @@ impl<'db> FunctionType<'db> {
         ))
     }
 
+    /// Convert the `FunctionType` into a [`Type::BoundMethod`].
+    pub(crate) fn into_bound_method_type(
+        self,
+        db: &'db dyn Db,
+        self_instance: Type<'db>,
```

Comment:
The parameter name is a bit misleading because in this PR above where this is being called it's not an instance of the class but the class itself. This is also the case for the field name on `BoundMethodType` which is also called `self_instance`. I'm not sure what to name it, maybe `self_or_cls` (and remove the `_instance` suffix).

Regardless, this shouldn't block this PR and something that could be done in a follow-up instead.